### ISSUE TITLE
Add an optional timezone parameter to `dateadd`.

### ISSUE THREAD

**Issue by peter-facko** [questdb/questdb#5114]
### Is your feature request related to a problem?

Using `dateadd` works in other timezones than UTC, but we need to convert to the timezone and back and so the code is too lengthy.

### Describe the solution you'd like.

An optional timezone parameter added to `dateadd`.

```SQL
DATEADD('w', 1, '2024-10-21', 'Europe/Bratislava')
```

### Describe alternatives you've considered.

```SQL
TO_UTC(DATEADD('w', 1, TO_TIMEZONE('2024-10-21', 'Europe/Bratislava')), 'Europe/Bratislava')
```

### Full Name:

Peter Fačko

### Affiliation:

myself

**Comment by abant07** [comment]
Hey @nwoolmer 

I was wondering if I could take up this issue if you think of it would be better that the questdb internal team takes up #5098 instead?

**Comment by abant07** [comment]
Great! Will do

**Comment by abant07** [comment]
Hey @peter-facko 

Just to clarify, are you asking for a DateAdd function that will take in a UTC time, and add a certain period to it (week, day, month, etc)? However, the problem you are running into now is that Dateadd only accepts times from local timezones?

**Comment by peter-facko** [comment]
Yes, you understand correctly. It should be equivalent to the second piece of code I provided.

And yes, DST is the problem here, as the example shows. There is a one hour difference when you don't apply the timezone conversions.

**Comment by bluestreak01** [comment]
hi @peter-facko i know i am late to the party here. Could you give us an example where `dataadd` will work differently in UTC and any given timezone?

For anything outside of "month" and "year" the conversion is just a simple addition. E.g. 
```
new_timestamp = old_timestamp + micros
```

if you're doing in in timezone, the formula becomes

```
new_timestamp + offset = old_timestamp + offset + micros
```

As you can see converting timestamp to a timezone before or after `dateadd` does not move a needle.

I am referring to this formula specifically because your example is "weeks", which is 7 day in microseconds.

The `dateadd` becomes slightly more involved for "months" and "years", but I still cannot see how moving timestamp to timezone before `dateadd` changes things there.

**Comment by abant07** [comment]
I believe the concern here is daylight savings time.

@peter-facko would like a function that is able to add time to a timestamp using a particular local zone as reference for DST. The new utc date should take into consideration if the new time should be pushed back an hour or brought forward an hour

Previously, I believe the other dateadd function did not take this into account, and so the utc time zone would be off by 1. His temporary solution was first converting from utc to local, then adding the time, then converting back to utc.

**Comment by abant07** [comment]
His example of adding a week to 10/21/2024 is an example of the DST off by 1

**Comment by bluestreak01** [comment]
what i am essentially saying is that `dateadd` fits this functional equation:
```
f(x + y) = f(x) + y
```
Here:
`f` - is `dateadd` function
`x` - is UTC timestamp
`y` - it timezone offset

If the intent is to display UTC time in a particular timezone, it can be done using

```
to_timezone(timestamp, 'Europe/Bratislava')
```

so instead of using
```
TO_UTC(DATEADD('w', 1, TO_TIMEZONE('2024-10-21', 'Europe/Bratislava')), 'Europe/Bratislava')
```

it could be```
```
to_timezone(DATEADD('w', 1, '2024-10-21'), 'Europe/Bratislava')
```

**Comment by bluestreak01** [comment]
@abant07 you are correct, thank you
